src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +107,10 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(c -> new CandidateValidator().test(c))
```

Comment:
You've already defined logic in you CandidateValidator, so that it`s not necessary to call that method
```suggestion
                .filter(new CandidateValidator())
```